src/theory/quantifiers/sygus/sygus_interpol.h

```diff
@@ -0,0 +1,179 @@
+ * conjecture that encodes an interpolation problem. In detail, if our input
+ * formula is F( x ) for free symbols x, and is partitioned into axioms Fa and
+ * conjecture Fc then the sygus conjecture we construct is:
+ *
+ * exists A. forall x. ( (Fa( x ) => A( x )) ^ (A( x ) => Fc( x )) )
+ *
+ * where A( x ) is a predicate over the free symbols of our input that are
+ * shared between Fa and Fc. In other words, A( x ) must be implied by our
+ * axioms Fa and implies Fc( x ).
+ */
+class SygusInterpol
+{
+ public:
+  SygusInterpol();
+
+  SygusInterpol(LogicInfo logic);
+
+  /**
+   * Returns the sygus conjecture corresponding to the interpolation problem for
+   * input problem (F above) given by axioms (Fa above), and conj (Fc above).
+   *  Note that axioms is expected to be a subset of asserts.
+   *
+   * The argument name is the name for the interpol-to-synthesize.
+   *
+   * The type itpGType (if non-null) is a sygus datatype type that encodes the
+   * grammar that should be used for solutions of the interpolation conjecture.
+   *
+   * The output argument interpol is the solution to the synthesis conjector.
+   *
+   * The relationship between the free variables of asserts and the formal
+   * argument list of the interpol-to-synthesize are tracked by the attribute
+   * SygusVarToTermAttribute.
+   *
+   * In particular, solutions to the synthesis conjecture will be in the form
+   * of a closed term (lambda varlist. t). The intended solution, which is a
+   * term whose free variables are a subset of asserts, is the term
+   * t * { varlist -> SygusVarToTermAttribute(varlist) }.
+   */
+  bool SolveInterpolation(const std::string& name,
+                          const std::vector<Node>& axioms,
+                          const Node& conj,
+                          const TypeNode& itpGType,
+                          Expr& interpol);
+
+ private:
+  /**
+   * Collects symbols from axioms (axioms) and conjecture (conj), which stored
+   * in d_syms. And computes the shared symbols between axioms and conjecture,
+   * stored in d_symsShared.
+   */
+  void collectSymbols(const std::vector<Node>& axioms, const Node& conj);
+
+  /**
+   * Creates free variables and shared free variables from d_syms and
+   * d_symsShared, which stored in d_vars and d_varsShared. And also creates the
+   * same set of variables for formal argument list, which stored in d_vlvs and
+   * d_vlvsShared. Extracts the types of shared variables, which stored in
+   * d_varTypesShared. Creates the formal argument list of the
+   * interpol-to-synthesis, stored in d_abvlShared.
+   *
+   * The argument needsShared denotes if we want to restrict the argument list
+   * of the interpol-to-synthesis to be over shared variables or all the
+   * variables.
+   */
+  void createVariables(bool needsShared);
+
+  /**
+   * Set up the grammar for the interpol-to-synthesis.
+   *
+   * The user-defined grammar will be encoded by itpGType. In DEFAULT option, it
```

Comment:
Can you mention where this option is set (something like `options::smt::interpolationMode` or something like that?